tests/brevitas_examples/test_quantize_model.py

```diff
@@ -0,0 +1,305 @@
+import pytest
+from copy import deepcopy
+import torch
+import torch.nn as nn
+from brevitas.nn import QuantConv2d, QuantLinear, QuantReLU
+from brevitas_examples.imagenet_classification.ptq.ptq_common import quantize_model
+from brevitas.graph.quantize import preprocess_for_quantize
+
+import ipdb
+
+
+# CONSTANTS
+IMAGE_DIM = 16
+
+##################
+# EXAMPLE MODELS #
+##################
+@pytest.fixture
+def minimal_model():
+    """
+    Inputs:
+    Implictly takes in a torch.Tensor, size: (batch_size, 3, IMAGE_DIM, IMAGE_DIM).
+
+    Outputs:
+    Implictly returns a torch.Tensor, size: (batch_size, 16, IMAGE_DIM, IMAGE_DIM)
+
+    This model has 3 input channels. I.e., for `layerwise` quantization it will use `layerwise_first_last_bit_width`
+    bits for the weight and activation quantization.
+    """
+    return nn.Sequential(
+        nn.Conv2d(3, 16, kernel_size=3, padding=1),
+        nn.ReLU(),
+    )
+
+@pytest.fixture
+def simple_model():
+    """
+    Inputs:
+    Implictly takes in a torch.Tensor, size: (batch_size, 10, IMAGE_DIM, IMAGE_DIM).
+
+    Outputs:
+    Implictly returns a torch.Tensor, size: (batch_size, 1000).
+
+    """
+    assert IMAGE_DIM % 2 == 0, "`IMAGE_DIM` should be a multiple of 2"
+    return nn.Sequential(
+        nn.Conv2d(10, 16, kernel_size=3, padding=1),
+        nn.ReLU(),
+        nn.Conv2d(16, 32, kernel_size=3, padding=1),
+        nn.ReLU(),
+        nn.MaxPool2d(2), # downsample from IMAGE_DIM to half that
+        nn.Flatten(),
+        nn.Linear(32 * int(IMAGE_DIM/2) ** 2, 1000)
+    )
+
+##############
+# Unit tests #
+##############
+@pytest.mark.parametrize("weight_bit_width", [2, 5, 8, 16])
+@pytest.mark.parametrize("act_bit_width", [2, 5, 8])
+@pytest.mark.parametrize("bias_bit_width", [16, 32])
+@pytest.mark.parametrize("layerwise_first_last_bit_width", [2, 8])
+def test_layerwise_10_in_channels_quantize_model(simple_model, weight_bit_width, bias_bit_width, act_bit_width, layerwise_first_last_bit_width):
+    """
+    We use a model with  input channels, and test `layerwise` quantization.
+    Because of the 10 input channels, this will ignore the `layerwise_first_last_bit_width`
+    value, and will quantize evrything according to:
+    - weight_bit_width
+    - bias_bit_width
+    - act_bit_width
+    """
+    quant_model = quantize_model(
+        model=deepcopy(simple_model),
+        backend='layerwise',
+        weight_bit_width=weight_bit_width, 
+        act_bit_width=act_bit_width, 
+        bias_bit_width=bias_bit_width,
+        weight_quant_granularity='per_tensor',
+        act_quant_percentile=99.9,
+        act_quant_type='sym',
+        scale_factor_type='float_scale',
+        quant_format='int',
+        layerwise_first_last_bit_width=layerwise_first_last_bit_width,
+        #layerwise_first_last_mantissa_bit_width=10,
```

Comment:
Why are these commented?

Reply:
No good reason, this is just a WIP that I wanted to expand on. It also has `ipdb` imported for the same reason. Before merging I will remove both.

@Giuseppe5  Are there any specific tests you would like to see? I was going to just play around and generally increase test coverage for this a bit more.